idb/postgres/internal/ledger_for_evaluator/ledger_for_evaluator.go

```diff
@@ -388,96 +376,101 @@ func (l *LedgerForEvaluator) loadAccounts(addresses map[basics.Address]struct{})
 		return nil, fmt.Errorf("loadAccounts() err: %w", err)
 	}
 
+	// Add special accounts if needed.
+	for address := range addresses {
+		if l.isSpecialAddress(address) {
+			// The balance of a special address must pass the minimum balance check in
+			// go-algorand's evaluator, so return a sufficiently large balance.
+			var balance uint64 = 1000 * 1000 * 1000 * 1000 * 1000
+			accountData := new(basics.AccountData)
+			*accountData = basics.AccountData{
+				MicroAlgos: basics.MicroAlgos{Raw: balance},
+			}
+			res[address] = accountData
+		}
+	}
+
 	return res, nil
 }
 
-// PreloadAccounts loads the account data for the given addresses and stores them
-// in the internal cache.
-func (l *LedgerForEvaluator) PreloadAccounts(addresses map[basics.Address]struct{}) error {
-	accountData, err := l.loadAccounts(addresses)
+func (l *LedgerForEvaluator) parseAddress(row pgx.Row) (basics.Address, bool /*exists*/, error) {
+	var buf []byte
+	err := row.Scan(&buf)
+	if err == pgx.ErrNoRows {
+		return basics.Address{}, false, nil
+	}
 	if err != nil {
-		return err
+		return basics.Address{}, false, fmt.Errorf("parseAddress() err: %w", err)
 	}
 
-	l.preloadedAccountData = accountData
-	return nil
+	var address basics.Address
+	copy(address[:], buf)
+
+	return address, true, nil
 }
 
-// LookupWithoutRewards is part of go-algorand's ledgerForEvaluator interface.
-func (l LedgerForEvaluator) LookupWithoutRewards(round basics.Round, address basics.Address) (basics.AccountData, basics.Round, error) {
-	// The balance of a special address must pass the minimum balance check in
-	// go-algorand's evaluator, so return a sufficiently large balance.
-	if l.isSpecialAddress(address) {
-		var balance uint64 = 1000 * 1000 * 1000 * 1000 * 1000
-		accountData := basics.AccountData{
-			MicroAlgos: basics.MicroAlgos{Raw: balance},
-		}
-		return accountData, round, nil
+// GetAssetCreator is part of go-algorand's indexerLedgerForEval interface.
+func (l LedgerForEvaluator) GetAssetCreator(indices map[basics.AssetIndex]struct{}) (map[basics.AssetIndex]ledger.FoundAddress, error) {
```

Comment:
Was there any thought about organizing this so that it could be batched along with the account lookup?

Reply:
What about the query duration? Sounds like the RTT isn't really a factor, but running each of these in parallel could have some value.

Reply:
I don't know if we can influence parallelism and gain performance. One thing that comes to mind is read requests going to different read replicas of aurora can help, but we don't have control over it, and I'm not sure the load balancer is smart enough. In addition, other databases (regular postgres) might have worse performance because of parallelism (one thing that comes to mind is HDD where parallel reading causes too many head jumps so the overall performance is worse, not sure how it works for SSD). Also, if we want to parallelize, we should start with parallelizing account query from the `account` table, as that is longest part of reading data.